scripts/update_dep.sh

```diff
@@ -1,33 +1,67 @@
 #!/bin/bash 
 
 # Usage:
-#    ./scripts/update_dep.sh module version
-# or ./scripts/update_dep.sh module
+#    ./scripts/update_dep.sh should_force_upgrade module version
```

Comment:
Suggest not to introduce the additional parameter `should_force_upgrade` to keep the script as simple as possible. If we don't want to bump pure indirect dependency, then we shouldn't run this script at all.

Reply:
I totally understand the concern. 

The only reason I did this is because of the use case (documented in the PR description https://github.com/etcd-io/etcd/pull/18609#issue-2536917519) where one dependency bump caused version inconsistency, thus, forcing us to bump the pure indirect dependency. 

I can remove this feature easily, but I would like to bring the context and confirm with you that we would like to drop this argument :) 

Thanks

Reply:
Probably I missed something. I was just suggesting to remove the parameter `should_force_upgrade`, but we still need to make sure each dependency's version is consistent across all modules.

Reply:
I don't think you missed anything, it's just that I added `should_force_upgrade` to support the case where we are forced to bump the fully indirect dependency when we are forced to (which is not what we usually do).

I will remove `should_force_upgrade` for now! :)